feat: Add option to bypass subscriptions creation

Roming22 opened the pull request:
Clusters might already have subscriptions installed, which would trigger an error in Helm as it did not control the resources.

A new flag is added to disable the subscription installation per product, with the following limitations:
- AMQ Streams is controlled by TPA
- CrunchyData is controlled by RHDH and TPA

Those limitations means that a user may need to install some subscription manually (eg if they already have AMQ Streams, TPA must be installed manually and the `managedSubscription` flag must be set to `false`).

cf [RHTAP-4063](https://issues.redhat.com//browse/RHTAP-4063)

rh-pre-commit.version: 2.3.2
rh-pre-commit.check-secrets: ENABLED

otaviof commented on installer/charts/rhtap-subscriptions/values.yaml:
```diff
@@ -22,7 +22,7 @@ subscriptions:
     description: OpenShift GitOps Operator
     apiResource: gitopsservices.pipelines.openshift.io
-    namespace: openshift-operators
+    namespace: openshift-gitops-operator
```
Interesting. Back in the days, I had issues on moving subscription operators outside of `openshift-operators` namespace.

Roming22 replied:
I've changed it because it's the default namespace used if you install the subscription from the console. I thought it would be a good idea to be as vanilla as possible.

Roming22 replied:
I didn't test it properly. You are right, it doesn't work as OCP does not allow the creation of `openshift-*` namespaces.

I'll revert this change.

Roming22 replied:
Reverted.

otaviof reviewed (approved):
Nice one! I liked the approach exploring the feature properties dictionary 👍

/lgtm

Roming22 commented:
/retest

Roming22 committed:
feat: Add option to bypass subscriptions creation

Clusters might already have subscriptions installed, which would trigger
an error in Helm as it did not control the resources.

A new flag is added to disable the subscription installation per
product, with the following limitations:
- AMQ Streams is controlled by TPA
- CrunchyData is controlled by RHDH and TPA

Those limitations means that a user may need to install some
subscription manually (eg if they already have AMQ Streams, TPA must be
installed manually and the `managedSubscription` flag must be set to
`false`).

cf RHTAP-4063

rh-pre-commit.version: 2.3.2
rh-pre-commit.check-secrets: ENABLED

Roming22 commented:
/retest

Roming22 commented:
/retest

xinredhat committed:
Merge branch 'main' into feat/subscriptions

Roming22 commented:
/retest

xinredhat commented:
/retest

xinredhat commented:
/lgtm
